nilearn/experimental/datasets/_datasets.py

```diff
@@ -0,0 +1,147 @@
+    FileMesh,
+    Mesh,
+    PolyMesh,
+    SurfaceImage,
+)
+
+
+def load_fsaverage(
+    mesh_name: str = "fsaverage5",
+) -> dict[str, PolyMesh]:
+    """Load fsaverage for both hemispheres."""
+    fsaverage = datasets.fetch_surf_fsaverage(mesh_name)
+    renaming = {
+        "pial": "pial",
+        "white": "white_matter",
+        "infl": "inflated",
+        "sphere": "sphere",
+        "flat": "flat",
+    }
+    meshes: dict[str, dict[str, Mesh]] = {
+        mesh_name: {
+            f"{hemisphere}": FileMesh(fsaverage[f"{mesh_type}_{hemisphere}"])
+            for hemisphere in ("left", "right")
+        }
+        for mesh_type, mesh_name in renaming.items()
+    }
+    renaming = {"curv": "curvature", "sulc": "sulcal", "thick": "thickness"}
+    return meshes
+
+
+ALLOWED_DATA_TYPES = (
+    "curvature",
+    "sulcal",
+    "thickness",
+)
+
+
+def load_fsaverage_data(
+    mesh_name: str = "fsaverage5",
+    mesh_type: str = "pial",
+    data_type: str = "sulcal",
+) -> SurfaceImage:
+    """Return freesurfer data on an fsaverage mesh."""
+    if mesh_type not in ALLOWED_MESH_TYPES:
+        raise ValueError(
+            f"'mesh_type' must be one of {ALLOWED_MESH_TYPES}.\n"
+            f"Got: {mesh_type}."
+        )
+    if data_type not in ALLOWED_DATA_TYPES:
+        raise ValueError(
+            f"'data_type' must be one of {ALLOWED_DATA_TYPES}.\n"
+            f"Got: {data_type}."
+        )
+
+    fsaverage = load_fsaverage(mesh_name)
+    old_fsaverage = datasets.fetch_surf_fsaverage(mesh_name)
+
+    renaming = {"curvature": "curv", "sulcal": "sulc", "thickness": "thick"}
+    img = SurfaceImage(
+        mesh=fsaverage[mesh_type],
+        data={
+            "left": _io.read_array(
+                old_fsaverage[f"{renaming[data_type]}_left"]
+            ),
+            "right": _io.read_array(
+                old_fsaverage[f"{renaming[data_type]}_right"]
+            ),
+        },
+    )
+
+    return img
+
+
+ALLOWED_MESH_TYPES = (
+    "pial",
+    "white_matter",
+    "inflated",
+    "sphere",
+    "flat",
+)
+
+
+def fetch_nki(mesh_type: str = "pial", **kwargs) -> Sequence[SurfaceImage]:
+    """Load NKI enhanced surface data into a surface object."""
+    if mesh_type not in ALLOWED_MESH_TYPES:
+        raise ValueError(
+            f"'mesh_type' must be one of {ALLOWED_MESH_TYPES}.\n"
+            f"Got: {mesh_type}."
+        )
+
+    fsaverage = load_fsaverage("fsaverage5")
+
+    nki_dataset = datasets.fetch_surf_nki_enhanced(**kwargs)
+
+    images = []
+    for left, right in zip(
+        nki_dataset["func_left"], nki_dataset["func_right"]
+    ):
+        left_data = _io.read_array(left).T
+        right_data = _io.read_array(right).T
+        img = SurfaceImage(
+            mesh=fsaverage[mesh_type],
+            data={
+                "left": left_data,
+                "right": right_data,
+            },
+        )
+        images.append(img)
+
+    return images
+
+
+def fetch_destrieux(
```

Comment:
testing this will require improving the mocking of the dataset: would probably do this in another PR